src/shared/sam/activation.ts

```diff
@@ -165,3 +160,25 @@ async function activateCodeLensProviders(
 
     return disposables
 }
+
+class InlineDebugAdapterFactory implements vscode.DebugAdapterDescriptorFactory {
+    constructor(readonly ctx:ExtContext) {}
+
+    /**
+     * The inline implementation implements the Debug Adapter Protocol.
+     * VSCode's extension API has a minimalistic subset of that protocol:
+     *   - vscode.DebugAdapter.handleMessage(): for passing a DAP message to the adapter.
+     *   - vscode.DebugAdapter.onDidSendMessage(): for listening for DAP messages received from the adapter.
+     *
+     * - Alternative: import the "vscode-debugprotocol" node module.
+     * - Alternative (easier): use VSCode's default implementation of a debug
+     *   adapter, available as node module "vscode-debugadapter" in 1.38+ the
+     *   DebugSession (or LoggingDebugSession) is compatible with the
+     *   `vscode.DebugAdapter` interface defined in the extension API.
+     *
+     * https://code.visualstudio.com/updates/v1_42#_extension-authoring
+     */
```

Comment:
Interested by this: before, we just created a debug configuration and directly passed it into an existing debugger (e.g. Node, Python, .NET). Is there a reason why we should move away from that model, and what ramifications does this have in terms of adding new runtimes?

Reply:
latest commit actually moves away from this. This was the DAP approach which I later found is probably not necessary for our use-case.

Though it might be worth keeping the skeleton, it's harmless and could be useful later for handling various debugger events.